orm/src/main/java/org/hibernate/search/batchindexing/impl/MassIndexerImpl.java

```diff
@@ -78,16 +79,19 @@ protected MassIndexerImpl(SearchIntegrator searchIntegrator, SessionFactoryImple
 	 * @return a new set of entities
 	 */
 	private static IndexedTypeSet toRootEntities(ExtendedSearchIntegrator extendedIntegrator, Class<?>... selection) {
-		Set<Class<?>> entities = new HashSet<Class<?>>();
+		IndexedTypeSet selectionAsSet = IndexedTypeSets.fromClasses( selection );
 		//first build the "entities" set containing all indexed subtypes of "selection".
-		for ( Class<?> entityType : selection ) {
-			IndexedTypeSet targetedClasses = extendedIntegrator.getIndexedTypesPolymorphic( IndexedTypeSets.fromClass( entityType ) );
-			if ( targetedClasses.isEmpty() ) {
-				String msg = entityType.getName() + " is not an indexed entity or a subclass of an indexed entity";
-				throw new IllegalArgumentException( msg );
+		Set<Class<?>> entities = extendedIntegrator.getIndexedTypesPolymorphic( selectionAsSet ).toPojosSet();
```

Comment:
This refactoring isn't semantically equivalent. Was that intentional?

Reply:
Done, it should be better now.